lib/Base/System/ClientState.vala

```diff
@@ -191,13 +191,32 @@ public partial class Backend.Client : Object {
   private async Session unpack_session (Variant variant) throws Error {
     string? uuid_prop, platform_name, server_prop,
             username_prop, access_prop;
+    bool auto_start_prop;
     PlatformEnum platform_prop;
 
     // Attempt to load the server data
-    variant.lookup ("uuid", "ms", out uuid_prop);
-    variant.lookup ("platform", "ms", out platform_name);
-    variant.lookup ("server_uuid", "ms", out server_prop);
-    variant.lookup ("username", "ms", out username_prop);
+    if (variant.get_type_string () == "a{sms}") {
+      variant.lookup ("uuid", "ms", out uuid_prop);
+      variant.lookup ("platform", "ms", out platform_name);
+      variant.lookup ("server_uuid", "ms", out server_prop);
+      variant.lookup ("username", "ms", out username_prop);
+      auto_start_prop = true;
+    }
+    else {
+      Variant? uuid_variant, platform_name_variant, server_variant, username_variant, auto_start_variant, geometry_variant;
+      // Newer format `a{smv}`
+      variant.lookup ("uuid", "mv", out uuid_variant);
+      variant.lookup ("platform", "mv", out platform_name_variant);
+      variant.lookup ("server_uuid", "mv", out server_variant);
+      variant.lookup ("username", "mv", out username_variant);
+      variant.lookup ("auto_start", "mv", out auto_start_variant);
+      uuid_prop = uuid_variant.get_string();
+      platform_name = platform_name_variant.get_string();
+      server_prop = server_variant.get_string();
+      username_prop = username_variant.get_string();
+      auto_start_prop = auto_start_variant.get_boolean();
+    }
```

Comment:
Since we're still in an alpha here, I would say we can just replace the old format without compatibility layer.

Reply:
Fair enough. I was trying to avoid having to log in _again_. But since I've migrated my accounts and you're happy to lose yours then I can simplify that 🙂